### PR TITLE
Updated custom-route docs post 9.5 Release

### PR DESCRIPTION
**Pull request by jaymarvels**
Update custom route docs to include more succinct information about utilising custom routes and routes that are considered client side requests since 9.5 was released that fixes some previously noted bugs

Profile: https://our.umbraco.com/members/id:294245/

### CODE REVIEW

**Comment by eshanrnh** [comment]
Lasse's team will review it in the next sprint.

File: Reference/Routing/custom-routes.md [hunk 1]
```diff
@@ -300,6 +296,66 @@ namespace RoutingDocs.Controllers
...
+```
+public IPublishedContent FindContent(ActionExecutingContext actionExecutingContext)
+{
+    IPublishedContent content = default(IPublishedContent);
+    var context = _umbracoContextAccessor.GetRequiredUmbracoContext();
+
+    content = context.Content.GetAtRoot().FirstOrDefault();
+
+    return content;
+}
```
Comment (nikolajlauridsen) [inline Reference/Routing/custom-routes.md:352]:
```suggestion
public IPublishedContent? FindContent(ActionExecutingContext actionExecutingContext)
{
    IUmbracoContext context = _umbracoContextAccessor.GetRequiredUmbracoContext();
    IPublishedContent? content = context.Content?.GetAtRoot().FirstOrDefault();

    return content;
}
```

I'd recommend changing this method up a bit to something like this. 

This way these docs will work for both v9 and v10, even though v9 complains with "The annotation for nullable reference types should only be used in code within a '#nullable' annotations context", however this works just fine still. 

Fun bonus fact, reviewing this showed us a nullability bug in v10 that I've fixed here: https://github.com/umbraco/Umbraco-CMS/pull/12545/ so double H5YR :D

**Review by nikolajlauridsen — COMMENTED** [review]
Overall looks good to me, easy to understand and works just fine 👍 💪 

I've added one small suggestion, simplifying the example code a bit and making it compatible with both V9 and V10 😄

File: Reference/Routing/custom-routes.md (hunk 1 above)
Comment (jaymarvels) [reply Reference/Routing/custom-routes.md:352]: LGTM and happy with the changes if it makes sense for v10 docs

File: Reference/Routing/custom-routes.md [hunk 2]
```diff
@@ -300,6 +296,64 @@ namespace RoutingDocs.Controllers
 
 With that we have our controller with a custom route within an Umbraco context.
 
+#### Client Side Requests
+If the endpoint of your custom route is considered a client-side request e.g. **/sitemap.xml**, you will need make a few of changes to get this to work.
```
Comment (eshanrnh) [inline Reference/Routing/custom-routes.md:300]:
```suggestion
If the endpoint of your custom route is considered a client-side request e.g. **/sitemap.xml**, you will need to make a few changes to get this to work.
```

File: Reference/Routing/custom-routes.md [hunk 3]
```diff
@@ -300,6 +296,64 @@ namespace RoutingDocs.Controllers
 
 With that we have our controller with a custom route within an Umbraco context.
 
+#### Client Side Requests
```
Comment (eshanrnh) [inline Reference/Routing/custom-routes.md:299]:
```suggestion
#### Client-Side Requests
```

File: Reference/Routing/custom-routes.md [hunk 4]
```diff
@@ -300,6 +296,64 @@ namespace RoutingDocs.Controllers
...
+{
+    IUmbracoContext context = _umbracoContextAccessor.GetRequiredUmbracoContext();
+    IPublishedContent? content = context.Content?.GetAtRoot().FirstOrDefault();
+
+    return content;
+}
+```
+
+:::note
+There is a currently a bug in all versions below 9.5, where this fix won't work for mapping a client-side request to an Umbraco Controller. See [https://github.com/umbraco/Umbraco-CMS/issues/12083](https://github.com/umbraco/Umbraco-CMS/issues/12083) for more details. v9.5 fixes this issue and its recommended to update to the latest version!
```
Comment (eshanrnh) [inline Reference/Routing/custom-routes.md:354]:
```suggestion
There is currently a bug in all versions below 9.5, where this fix won't work for mapping a client-side request to an Umbraco Controller. See [https://github.com/umbraco/Umbraco-CMS/issues/12083](https://github.com/umbraco/Umbraco-CMS/issues/12083) for more details. v9.5 fixes this issue and it's recommended to update to the latest version!
```

**Review by eshanrnh — APPROVED** [review]
Looks good, @jaymarvels 💪 Merging the PR in.

**Comment by eshanrnh** [comment]
Woohoo! You now have a new C-Trib badge on your Our profile 🎉
Thanks again for helping. Looking forward to seeing more contributions from you in the future 💪

**Pull request merged**